eoc_journal/eoc_journal.py

```diff
@@ -2,6 +2,12 @@
 An XBlock that allows learners to download their activity after they finish their course.
 """
 
+from collections import defaultdict
+
+from opaque_keys.edx.keys import CourseKey
+from problem_builder.models import Answer
+from problem_builder.step import _normalize_id
```

Comment:
We should not be importing private methods that start with `_`. Since this is a small and simple method, I think we can just duplicate it in this repo.